lib/Parse/ParseDecl.cpp

```diff
@@ -5741,6 +5747,17 @@ void Parser::ParseFunctionDeclarator(Declarator &D,
     }
   }
 
+  if (getLangOpts().CheckedC && Tok.is(tok::colon)) {
```

Comment:
>if (getLangOpts().CheckedC && Tok.is(tok::colon)) { [](start = 2, length = 51)

I know you called it out in the header comment for this function, but a quick brief comment calling out that this is for the return bounds expression might be nice (similar to the ones in the few code chunks directly above).